src/query/api/v1/middleware/rewrite.go

```diff
@@ -117,8 +124,37 @@ func rewriteRangeDuration(
 	if err != nil {
 		return err
 	}
-
-	attrs, err := store.QueryStorageMetadataAttributes(ctx, params.start, params.end, fetchOpts)
+	startTime, endTime := params.start, params.end
+
+	// Using the prometheus engine in this way should be considered
+	// optional and best effort. Fall back to the frequently accurate logic
+	// of using the start and end time in the request
+	if opts.PrometheusEngineFn != nil {
+		lookback := opts.DefaultLookback
+		if params.isLookbackSet {
+			lookback = params.lookback
+		}
+		engine, err := opts.PrometheusEngineFn(lookback)
```

Comment:
You may want to add a TODO similar to the one here: https://github.com/m3db/m3/pull/3882/files#diff-0d8ef3d8555895061a7eb7f4b1e05852283ade888692425afe6db8cad09a4e17R66-R67

We have some hopes for the future to avoid creating a new engine each time we need to pass a lookback.